File: hw/viteri/u2/HW18ZooInPython/Zoo/model/Fish.py
from datetime import date
from model.Animal import Animal
from model.Cage import Cage

class Fish(Animal):
    def __init__(self, id, description, born_on_date, gender, cage, number_of_bones, vertebrate):
        super().__init__(id, description, born_on_date, gender, cage, number_of_bones, vertebrate)

    def feed(self, food_amount):
        raise NotImplementedError("Feeding not supported for Fish")

File: hw/segarra/u2/HW18ZooInPythonAndJavaScript/ec.edu.zooinjavascript.model/Fish.js
import Animal from "./Animal.js";

export default class Fish extends Animal{
    constructor(id, description, bornOnDate, gender, cage, numberOfBones, vertebrate){
        if (new.target === Fish) {
            throw new TypeError("Cannot construct Bird instances directly");
        }
        super(id, description, bornOnDate, gender, cage, numberOfBones, vertebrate)
    }
}